int32[] mic1
int32[] mic2
int32[] mic3
int32[] mic4